### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by andriy-shymkiv**

### CODE REVIEW

**Review by Anastasiia-Smirnova — APPROVED** [review]
Good job!

File: src/convertToObject.js [hunk 1]
```diff
@@ -3,16 +3,32 @@
...
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const object = {};
```
Comment (illiarerun) [inline src/convertToObject.js:17]:
```suggestion
```

File: src/convertToObject.js [hunk 2]
```diff
@@ -3,16 +3,32 @@
...
+  filtered.map(element => {
+    const splited = element.split(':');
+
+    if (splited.length === 2) {
+      const key = splited[0].trim();
+      const value = splited[1].trim();
+
+      object[key] = value;
+    }
+  });
```
Comment (illiarerun) [inline src/convertToObject.js:29]: don't use map to mutate anything, map is used to transform elements and create new array using them. use reduce to create new object

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/convertToObject.js [hunk 3]
```diff
@@ -3,16 +3,29 @@
...
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
-function convertToObject(sourceString) {
-  // write your code here
-}
+const convertToObject = (sourceString) => {
+  return sourceString.split(';')
```
Comment (denys-danyliuk) [inline src/convertToObject.js:17]:
```suggestion
  return sourceString
    .split(';')
```
4 rule in checklist

File: src/convertToObject.js [hunk 4]
```diff
@@ -3,16 +3,29 @@
...
+    .filter(item => item.includes(':'))
```
Comment (denys-danyliuk) [inline src/convertToObject.js:18]: this is redundant because of next line

**Review by denys-danyliuk — APPROVED** [review]

**Pull request closed without merging**